Sources/Purchasing/Purchases/PurchasesType.swift

```diff
@@ -1213,6 +1213,11 @@ internal protocol InternalPurchasesType: AnyObject {
     /// - Throws: `PublicError` if request failed.
     func healthRequest(signatureVerification: Bool) async throws
 
+    /// Asks the server for an in-depth report of the SDK's configuration
+    /// - Throws: A `BackendError` if the request fails for an invalid API key or for connectivity issues
+    /// - Returns: A health report with all checks performed in the server and their status
```

Comment:
let's add a bit more detail about _when_ you want to use this. 
Or maybe even keep it internal but automatically call it in debug builds? 

Otherwise, this could easily get called in a loop or like on every screen shown in a mega app

Reply:
100% Let me work on the comment a bit more. Unfortunately, since we have to use it in the Sample App, we're going to have to mark it as public, but we can definitely inform users of when they would like to use it.

In terms of running automatically on DEBUG builds, that is for sure the next step, I would like to try validate the approach in a less risky way through the debug menu and while we build the sample app, as we might need some tweaks here and there, but once we are done, let's do it!! 💪

Reply:
Looking at this type it looks like it is already internal, but I think we are both referring to the methods in the `PurchasesDiagnostics` type.

I have updated the comments now and added a couple of `Important` notes to the documentation:

<img width="658" alt="Screenshot 2025-04-11 at 11 36 22" src="https://github.com/user-attachments/assets/6eb1c9b6-f325-42fe-bba7-a501527e5add" />

Reply:
maybe we could also only leave it defined for debug builds to prevent this getting shipped in a mega app

Reply:
i.e.: wrapping it in `#if DEBUG`

Then we wouldn't even need the second part of that warning - the compiler would prevent it (although it's good to keep it for clarity)

Reply:
Oh yeah good point, done it now. I will leave the `Important` block just in case it is not immediate at first glance if people get failures on prod builds 👍